Dev edfi changes/v0.1

vir-neenopal opened the pull request:
Hi Team,

This branch has relevant commits to improve the Ed-Fi Module.

Please note that OEA base functions like upsert, overwrite, append, add_to_lake_db have also been modified to enable the implementation of the above changes. 

For more details,
1. Kindly refer to the README.md file in */Ed-Fi/notebook/ 
2. For EdFi_Refine notebook - kindly read the comments and acknowledgements added at the top of the notebook

Warm Regards,
Viraj Jayant
(Representing Edwire)

vir-neenopal committed:
Ed-Fi Module Changes

The changes are highlighted in the */Ed-Fi/notebook/README.md

edg-jf committed:
Merge branch 'microsoft:main' into dev-edfi-changes/v0.1

vir-neenopal committed:
Update EdFi_Refine.ipynb

vir-neenopal committed:
Update EdFi_Refine.ipynb

vir-neenopal committed:
Update EdFi_Refine.ipynb

vir-neenopal committed:
Update EdFi_Refine.ipynb

cviddenKwantum commented on framework/synapse/notebook/OEA_py.ipynb:
```diff
@@ -451,42 +451,105 @@
+        "                logger.info('Writing unpartitioned delta lake')\n",
+        "                df.write.format('delta').save(destination_url)\n",
+        "            elif partitioning and len(partitioning_cols) == 0:\n",
+        "                logger.info('Partitioning columns absent - defaulting to DistrictId and SchoolYear as partitioning columns')\n",
```
This looks like EdFi specific code and should belong in the edfi_py.ipynb module class. Please remove all such code from the OEA_py.ipynb notebook class.

cviddenKwantum reviewed (changes requested):
See code comments and please revise.

vir-neenopal committed:
Ed-Fi Specific changes isolation

vir-neenopal committed:
EdFi Pipeline Changes

1) API Urls to be fetched via a python notebook
2) Corresponding notebook referenced
3) Dependent changes on the pipeline
4) OEA_py indentation correction

vir-neenopal committed:
Pipeline Name + Dir Edits

vir-neenopal committed:
Main Pipeline Name + Dir changes

vir-neenopal committed:
/basic added in directories

vir-neenopal committed:
Notebook References Changed

vir-neenopal committed:
Code correction to use OEA's pseudonymize correctly

vir-neenopal committed:
Revision to exclude irrelevant columns from df_lookup

vir-neenopal committed:
Indent Correction

vir-neenopal committed:
Indent Correction

vir-neenopal committed:
Declareed apiVersion in code

edg-jf committed:
Provide support for setup.sh input arguments
Adding support for custom resource group name

edg-jf committed:
Merge branch 'dev-edfi-changes/v0.1' of https://github.com/EdWire/OpenEduAnalytics into dev-edfi-changes/v0.1

vir-neenopal committed:
Ed-Fi Module Set-up change

This commit also ensures that the edfi_py and edfi_fetch_url notebooks are also installed by the setup bash script

vir-neenopal committed:
Merge branch 'dev-edfi-changes/v0.1' of https://github.com/EdWire/OpenEduAnalytics into dev-edfi-changes/v0.1

vir-neenopal committed:
Ed-Fi Assets Debugged + Other Edits

1. Debugged bugs hindering runs of codes
2. Made changes to pipeline orchestration to correctly pass parameters
3. Made changes to the edfi landEntity utility to work with test data as well (when no prod data is provided)

vir-neenopal committed:
changed state to active

vir-neenopal committed:
LastModifiedDate Added

vir-neenopal committed:
LastModifiedDate - try / except

Inclusion of LastModifiedDate is under testing - for now, included within try / except clause

vir-neenopal commented:
[Changes Notes - 2023-10-27](https://docs.google.com/document/d/1n-If5ScrDTPVZpjmgN3Bq0A2uEzq7o5zssaoM-H1gZE/edit?usp=sharing)

edg-jf committed:
Adding script argument in content of Readme
Reverting to CRLF line carriage return

edg-jf committed:
Merge branch 'dev-edfi-changes/v0.1' of https://github.com/EdWire/OpenEduAnalytics into dev-edfi-changes/v0.1
